server/opts.go

```diff
@@ -4810,6 +4814,42 @@ func parseWebsocket(v interface{}, o *Options, errors *[]error, warnings *[]erro
 			o.Websocket.TokenCookie = mv.(string)
 		case "no_auth_user":
 			o.Websocket.NoAuthUser = mv.(string)
+		case "headers":
+			tk, mv = unwrapValue(mv, &lt)
+			mvv, ok := mv.(map[string]interface{})
+			if !ok {
+				err := &configErr{tk, fmt.Sprintf("error parsing headers: unsupported type %T", mv)}
+				*errors = append(*errors, err)
+			}
+			m := make(map[string]string)
+			for kk, kv := range mvv {
+				_, v = unwrapValue(kv, &tk)
+				m[kk] = v.(string)
```

Comment:
Would make sure that we have key/value string pairs here. I also added a missing `continue` when processing handshake_timeout (harmless since we initialize the handshake to 0, so the missing continue has no effect, but just for correctness). Here is a diff of opts.go:
```
diff --git a/server/opts.go b/server/opts.go
index bfae98f5..0ba2a360 100644
--- a/server/opts.go
+++ b/server/opts.go
@@ -4794,6 +4794,7 @@ func parseWebsocket(v interface{}, o *Options, errors *[]error, warnings *[]erro
                        default:
                                err := &configErr{tk, fmt.Sprintf("error parsing handshake timeout: unsupported type %T", mv)}
                                *errors = append(*errors, err)
+                               continue
                        }
                        o.Websocket.HandshakeTimeout = ht
                case "compress", "compression":
@@ -4820,11 +4821,21 @@ func parseWebsocket(v interface{}, o *Options, errors *[]error, warnings *[]erro
                        if !ok {
                                err := &configErr{tk, fmt.Sprintf("error parsing headers: unsupported type %T", mv)}
                                *errors = append(*errors, err)
+                               continue
                        }
                        m := make(map[string]string)
                        for kk, kv := range mvv {
                                _, v = unwrapValue(kv, &tk)
-                               m[kk] = v.(string)
+                               if vv, ok := v.(string); ok {
+                                       m[kk] = vv
+                               } else {
+                                       err := &configErr{tk, fmt.Sprintf("headers content should be key/value strings, got %T", v)}
+                                       *errors = append(*errors, err)
+                                       break
+                               }
+                       }
+                       if len(*errors) > 0 {
+                               continue
                        }
 
                        var sb strings.Builder
```